test/e2e/specs/editor/various/autocomplete-and-mentions.spec.js

```diff
@@ -98,11 +100,28 @@ test.describe( 'Autocomplete', () => {
 <!-- /wp:paragraph -->`;
 			}
 
-			await page.click( 'role=button[name="Add default block"i]' );
+			await editor.canvas.click(
+				'role=button[name="Add default block"i]'
+			);
 			await page.keyboard.type( testData.triggerString );
 			await expect(
 				page.locator( `role=option[name="${ testData.optionText }"i]` )
 			).toBeVisible();
+			const ariaOwns = await editor.canvas.evaluate( () => {
+				return document.activeElement.getAttribute( 'aria-owns' );
+			} );
+			const ariaActiveDescendant = await editor.canvas.evaluate( () => {
+				return document.activeElement.getAttribute(
+					'aria-activedescendant'
+				);
+			} );
+			// Ensure `aria-owns` is part of the same document and ensure the
+			// selected option is equal to the active descendant.
+			await expect(
+				await editor.canvas
+					.locator( `#${ ariaOwns } [aria-selected="true"]` )
+					.getAttribute( 'id' )
+			).toBe( ariaActiveDescendant );
```

Comment:
This is probably a better fit in a unit test than an e2e test. Could it be done in a unit test environment giving the current architecture?

Reply:
Why should it be a unit test? We're checking if it's correct for an iframe because we need to duplicate the markup inside the iframe.

Reply:
It's just my preference since unit tests are faster and more reliable for this kind of task. I don't think we need anything from a real browser to perform this test. That said, it can still be tested in e2e tests if you prefer it though 🙂 .

Reply:
Here's some more context to why `Autocomplete` has mostly e2e tests instead of unit tests :)

Reply:
Thanks for the context! I would probably still prefer an integration test (rendering both `AutoComplete` and `RichText` without mocking), but nvm if it's too much trouble to set up.

Reply:
Agreed — we just went for what we thought was the best trade-off between required efforts and results.

Out of curiosity, where would those tests live in the repo?

Reply:
@glendaviesnz had some ideas about adding more integration tests to the repo. I haven't put many thoughts into it yet 😅 , so I'll just shamelessly defer this question to him 😆 .

Reply:
It is still on my to-do list to look at ways that we can more easily provide for  `integration` level testing for blocks, etc. rather than having to rely on e2e, eg. some way of testing the interaction between block inspector/toolbar/edit components without having to fire up a full WP instance. I am hoping to take more of a look at this once 6.2 is sorted.